pymesos/operator_v1.py

```diff
@@ -0,0 +1,558 @@
+                conn.request(method, path, body=data, headers=headers)
+                resp = conn.getresponse()
+            except Exception:
+                self._conn.close()
+                self._conn = None
+                raise
+
+            if resp.status < 200 or resp.status >= 300:
+                raise RuntimeError(
+                    'Failed to send request code=%s, message=%s' % (
+                        resp.status, resp.read()
+                    )
+                )
+
+            result = resp.read()
+            if not result:
+                return {}
+
+            try:
+                return json.loads(result.decode('utf-8'))
+            except Exception:
+                return {}
+
+    def getHealth(self):
+        body = dict(
+            type='GET_HEALTH',
+        )
+        return self._send(body)
+
+    def getFlags(self):
+        body = dict(
+            type='GET_FLAGS',
+        )
+        return self._send(body)
+
+    def getVersion(self):
+        body = dict(
+            type='GET_VERSION',
+        )
+        return self._send(body)
+
+    def getMetrics(self, timeout):
+        body = dict(
+            type='GET_METRICS',
+            get_metrics=dict(
+                timeout=dict(
+                    nanoseconds=timeout,
+                )
+            )
+        )
+        return self._send(body)
+
+    def getLoggingLevel(self):
+        body = dict(
+            type='GET_LOGGING_LEVEL',
+        )
+        return self._send(body)
+
+    def setLoggingLevel(self, level, duration):
+        body = dict(
+            type='SET_LOGGING_LEVEL',
+            set_logging_level=dict(
+                duration=dict(
+                    nanoseconds=duration,
+                ),
+                level=level,
+            )
+        )
+        return self._send(body)
+
+    def listFiles(self, path):
+        body = dict(
+            type='LIST_FILES',
+            list_files=dict(
+                path=path,
+            ),
+        )
+        return self._send(body)
+
+    def readFile(self, path, length, offset):
+        body = dict(
+            type='READ_FILE',
+            read_file=dict(
+                length=length,
+                offset=offset,
+                path=path,
+            )
+        )
+        return self._send(body)
+
+    def getState(self):
+        body = dict(
+            type='GET_STATE',
+        )
+        return self._send(body)
+
+    def getFrameworks(self):
+        body = dict(
+            type='GET_FRAMEWORKS',
+        )
+        return self._send(body)
+
+    def getExecutors(self):
+        body = dict(
+            type='GET_EXECUTORS',
+        )
+        return self._send(body)
+
+    def getTasks(self):
+        body = dict(
+            type='GET_TASKS',
+        )
+        return self._send(body)
+
+
+class MesosOperatorMasterDriver(Process, MesosOperatorDaemonDriver):
+    """
+        For now, supports only operations that retrieve data.
```

Comment:
Why? I found methods such like `createVolumes`, `setQuota` implemented. So we can update status of master as well as just retrieve data?

Reply:
It was an old forgotten comment. I deleted it. The Operator API is now 100% covered.